### ISSUE TITLE
Health API OAuth authorization error (invalid_scope)

### ISSUE THREAD

**Issue by willbach** [department-of-veterans-affairs/vets-api-clients#152]
## VA Lighthouse API Support Request

⚠️ Please read before submitting: As this is a public forum, please do not include any sensitive information (such as your VA API OAuth Client ID, VA API OAuth Client Secret, VA API key, Name, Address, Phone Number, Social Security Number, Medical Record Number, etc). For test apps, please make sure you are using your registered redirect URL, not the URL provided in the example.

## APIs affected (if applicable)

Health API

## How can we help?

I am getting an error when making the initial OAuth request (getting directed to the portal). The error is `OAuth authorization error (invalid_scope): One or more scopes are not configured for the authorization server resource.`

What is the proper scope I should be requesting for the sandbox?

Additionally, how do I provide a redirect url for the sandbox?

## What have you tried?

I have tried a few combinations of scopes:

`launch/patient patient/*.read offline_access`
`launch/patient patient/* offline_access`
`launch/patient offline_access patient/*specific record type*.read`

All of these give the same error. I have also tried omitting the scope in the request, but because it is the sandbox I get a message back saying that my application has no default scopes set.

## Additional context

None

**Comment by ryanlink** [comment]
Hi Will -

You provide a Redirect URI when requesting your OAuth credentials for sandbox.  Please do this at https://developer.va.gov/apply if you haven't already.

The documentation for OAuth flow is here:  
https://developer.va.gov/explore/health/docs/authorization

and the Health API:
https://developer.va.gov/explore/health/docs/fhir

If you have any further questions, or suggestions for updating our documentation, please let us know here or by emailing api@va.gov.

We also have a [Google Group](https://groups.google.com/forum/#!forum/va-lighthouse) discussion forum.

Cheers,
Ryan Link
Customer Success Manager
API Support Lead, Lighthouse | VA API Program

**Comment by willbach** [comment]
Thanks for the quick reply @ryanlink ! I was able to solve it with a little more trial and error. I had already specified my callback url on signup, so that wasn't relevant (my mistake).

(Leaving this here in case anyone encounters a similar issue)

I was attempting to connect to just the Health API and pull patient records. It is not possible to list all scopes under a wildcard like `patient/*.read`. The correct way to specify the scope is to list out all of the `patient` scopes like so: `launch/patient offline_access patient/Patient.read patient/AllergyIntolerance.read patient/Condition.read patient/CoverageEligibilityResponse.read patient/CommunityCareEligibility.read patient/DiagnosticReport.read patient/Immunization.read patient/Medication.read patient/MedicationOrder.read patient/MedicationStatement.read patient/Observation.read patient/Procedure.read`.
